pygama/dsp/_processors/multi_t_filter.py

```diff
@@ -0,0 +1,169 @@
+              "void(int32[:], int32[:], int32[:], int32, int32[:])",
+              "void(int64[:], int64[:], int64[:], int64, int64[:])"],
+             "(n),(m),(),()->(m)", nopython=True, cache=True)
+def multi_triggering_filter(wf_in, arbitary_list, threshold, delta, tp0_list):
+    """
+    Gets list of indices of the start of leading edges of multiple peaks within a waveform.
+    Is built to handle afterpulses/delayed cross talk and trains of pulses.
+    Parameters
+    ----------
+    wf_in : array-like
+        the array of data within which the list of tp0s will be found
+    arbitraty_list : array-like
+        A fixed length array of the same dimensions that you need the tp0 list to be
+        Used to tell numba what dimension of an array it should expect
+    threshold: scalar 
+        threshold to search for using time_point_thresh
+    delta : scalar
+        the absolute level by which data must vary (in one direction) about an
+        extremum in order for it to be tagged
+    Returns
+    -------
+    tp0_list : array-like
+        Array of fixed length (padded with nans) that hold the indices of
+        the identified initial rise times of peaks in the signal
+        
+    """
+     
+    # initialize arrays, padded with the elements we want
+    
+    tp0_list[:] = np.nan 
+    max_list= np.zeros_like(tp0_list)
+    min_list = np.zeros_like(tp0_list)
+    n_max = np.array([0],dtype=np.int32)
+    n_min = np.array([0],dtype=np.int32)
+    flag = np.array([0],dtype=np.int32)
+
+    # call njit_get_i_local_extrema to get locations of the maxima and minima within a waveform
+    
+    njit_get_i_local_extrema(wf_in, delta,max_list,min_list,n_max,n_min,flag)
+    
+    # Go through the list of maxima, calling njit_time_point_thresh on each maximum (ignoring the nan padding)
+    for i in range(int(n_max[0])):
+        if not np.isnan(max_list[i]): 
+            tp0_list[i] = njit_time_point_thresh(wf_in,float(threshold[0]),int(max_list[i]),tp0_list[i])
+            
+    """ 
+    time_point_thresh has issues with afterpulsing in waveforms that causes  
+    an aferpulse peak's tp0 to be sent to 0 or the same index as the tp0 for the first pulse.
+    This only happens when the relative minimum between the first pulse and 
+    the afterpulse is greater than the threshold. So, we sweep through the array again 
+    to ensure there are no duplicate indices. If there are duplicate indicies caused by a
+    misidentified tp0 of an afterpulse, we replace its index by that of the corresponding minimum
+    found using the njit_get_i_local_extrema function.
+    """
+
+    k=0
+    for index, i in enumerate(tp0_list):
+        for index2, j in enumerate(tp0_list[index+1:]):
+            if i == j:
+                tp0_list[index2+index+1] = min_list[k]
+        k+=1 # this makes sure that the index of the misidentified afterpulse tp0 is replaced with the correct corresponding minimum
+            
```

Comment:
In the end, what I think this function should end up looking like as an objectmode function is:
def multi_triggering_thresh(...):
    get_i_local_extrema(...)
    timepoint_thresh(...)
    remove_duplicates(...)

This way, we avoid any python loops that are slow in objectmode